crates/executor/src/lib.rs

```diff
@@ -0,0 +1,74 @@
+use std::future::Future;
+use std::mem;
+use std::sync::{Arc, Mutex};
+use std::task::{Context, Poll, Wake, Waker};
+
+use wasi::io;
+
+wit_bindgen::generate!({
+    world: "imports",
+    path: "io.wit",
+});
```

Comment:
nit: I personally prefer putting bindings into their own module (named something like `bindings`) so that it's easier to spot what is coming from from generated code vs. another module or crate. The way things are it makes it look like `wasi` is a 3rd party crate which can make understanding the code harder.

If you want `wasi` to be a part of the public API, then you can do `pub use bindings::wasi`. This just makes things much more explicit.